test/runtests.jl

```diff
@@ -78,6 +78,12 @@ end
         @time @safetestset "Ensemble solution statistics" begin
             include("downstream/ensemble_stats.jl")
         end
+        # @time @safetestset "Ensemble Optimization and Nonlinear problems" begin
+        #     include("downstream/ensemble_nondes.jl")
+        # end
```

Comment:
Is this intended?

Reply:
Yeah it was to show that other tests all pass, to get this one working it needs the dispatches we have been discussing

Reply:
I'm lost. What's the version that works with all tests?

Reply:
The one without type assertion in the __solve method

Reply:
Can you put it this in the form you believe works for everything and we can check the downstream tests and these tests?